### ISSUE TITLE
`!!Symbol.toStringTag` vs. `typeof Symbol.toStringTag === 'symbol'`  

### ISSUE THREAD

**Issue by justingrant** [inspect-js/is-date-object#22]
Should `!!Symbol.toStringTag` below instead be `typeof Symbol.toStringTag === 'symbol'` ?

https://github.com/inspect-js/is-date-object/blob/31c731c5efc5b1b86e6426d904373dc6225b929f/index.js#L15

I'm asking because there's odd behavior I'm seeing from either babel and/or webpack where sourcemaps are being emitted with code embedded in the sourcemap file that differs from the original source code. Below is output from a webpack plugin I wrote to compare sourcemap output to original source. https://github.com/webpack/webpack/issues/8302 may be related, although although this may be a different issue.  

```
Original Line 15: var hasToStringTag = typeof Symbol === 'function' && !!Symbol.toStringTag;
     Map Line 15: var hasToStringTag = typeof Symbol === 'function' && typeof Symbol.toStringTag === 'symbol';
```

Out of curiosity, which is the better code to use here?  The original source in this package or the updated code in the sourcemap? If the sourcemap is better, I'd be happy to PR the change here.

**Comment by ljharb** [comment]
Unfortunately no, it shouldn't, because core-js's symbol shams have the wrong typeof value for symbols, and if core-js has been ran, then Object.prototype.toString will respect Symbol.toStringTag, even if it's not a real symbol. (see 8943a4a5035b3f2c8cee9a5edabb55579c16983d)

Perhaps you have a version of is-date-object that's older than v1.0.4?

**Comment by justingrant** [comment]
Yep, that's exactly it. Some dependencies in my project are using 1.0.1, while others are using 1.0.4. So it's a de-duping issue, not a transpilation issue. Good to know, thanks. I'm not sure why the original source and the sourcemap are using different versions of the same code. Perhaps because all deps are asking for `^1.0.1` then webpack felt safe to de-dupe?  Admittedly, the de-duping parts of bundlers are mysterious to me. 

BTW, looks like the same pattern (`typeof Symbol.toStringTag === 'symbol'`) is present in [`is-callable`](https://github.com/inspect-js/is-callable/blob/32a52a9f0628199bff2393509534ae4a069a7420/index.js#L48). Does this also need to be updated?

**Comment by justingrant** [comment]
> BTW, looks like the same pattern (`typeof Symbol.toStringTag === 'symbol'`) is present in [`is-callable`](https://github.com/inspect-js/is-callable/blob/32a52a9f0628199bff2393509534ae4a069a7420/index.js#L48). Does this also need to be updated?

Also a bunch more places: https://github.com/search?q=org%3Ainspect-js+hasToStringTag&type=code

**Comment by ljharb** [comment]
Yeah, probably :-) i generally don’t try to make my code worse due to core-js, but when specific bugs are filed, i do.

**Comment by justingrant** [comment]
Lol. Do you want me to file issues across all those repos? From a quick search, looks like these are affected:
* is-typed-array
* which-builtin-type
* is-arguments
* is-generator-function
* is-arguments
* which-typed-array
* is-callable
* is-generator-function

**Comment by ljharb** [comment]
Check back in a day or five? I’ll probably make a common package that all of them can consume.

**Comment by ljharb** [comment]
Published https://npmjs.com/has-tostringtag; i'll update the relevant packages next.

**Comment by ljharb** [comment]
This issue is fixed for this package with 17a6df4; I think I've gotten to most of the others. If there's anything left, please feel free to comment here or file an issue on the relevant package.

**Comment by nspector** [comment]
Just filed this issue: https://github.com/inspect-js/which-typed-array/issues/51 Looks like this may be causing other issues
